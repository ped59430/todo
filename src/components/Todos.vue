<template>
  <div class="q-pt-xl q-mt-xl col-xs-12 col-sm-6 col-md-4">
    <div></div>
    <div class="title text-center q-pt-xl q-mt-xl">
      <span class="text-weight-medium">todo</span>toudou.
    </div>
    <div>
      <q-list separator>
        <todo-item
          v-for="(todo, key) in todos"
          :key="key"
          :uid="todo.uid"
          :description="todo.description"
          :done="todo.done"
        />
      </q-list>
      <q-item v-if="todos.length === 0" class="flex-center">
        <h5 class="text-grey-8">No todotoudous</h5>
      </q-item>
      <new-todo-item />
    </div>
  </div>
</template>

<script>
import TodoItemVue from "./TodoItem.vue";
import NewTodoItemVue from "./NewTodoItem.vue";

export default {
  name: "Todos",
  components: {
    "todo-item": TodoItemVue,
    "new-todo-item": NewTodoItemVue,
  },
  data() {
    return {};
  },
  computed: {
    todos() {
      return this.$store.state.todo.todos;
    },
  },
  methods: {
    getAllTodo() {
      this.$store.dispatch("todo/serverGetAllTodos");
    },
  },
  mounted() {
    this.getAllTodo();
  },
};
</script>

<style scoped>
.title {
  color: white;
  font-size: 2em;
}
</style>
