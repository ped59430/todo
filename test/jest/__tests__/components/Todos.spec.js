/* eslint-disable */
/**
 * @jest-environment jsdom
 */

import { mount, createLocalVue, shallowMount } from "@vue/test-utils";
import * as All from "quasar";
import Vuex from "vuex";
import Todos from "~/src/components/Todos.vue";
const { Quasar } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key];
  if (val && val.component && val.component.name != null) {
    object[key] = val;
  }
  return object;
}, {});

describe("Todos", () => {
  const localVue = createLocalVue();
  localVue.use(Quasar, { components });
  localVue.use(Vuex);

  let store;
  let mutations;
  let wrapper;
  let actions;
  const todo = {
    uid: 0,
    description: "zero",
    done: false
  };

  beforeAll(() => {
    mutations = {
      MUTATION: jest.fn()
    };
    actions = {
      serverEditTodo: jest.fn(),
      serverDeleteTodo: jest.fn(),
      serverGetAllTodos: jest.fn()
    };
    store = new Vuex.Store({
      modules: {
        todo: {
          namespaced: true,
          state: {
            todos: []
          },
          mutations,
          actions
        }
      }
    });
    wrapper = mount(Todos, {
      store,
      localVue,
      propsData: todo
    });
  });

  it("snapshot testing", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
