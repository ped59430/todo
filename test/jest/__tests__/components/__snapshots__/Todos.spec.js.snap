// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos snapshot testing 1`] = `
<div class="q-card q-mt-lg full-width" uid="0" description="zero">
  <div class="q-item q-item-type row no-wrap flex-center">
    <h1>Todos</h1>
  </div>
  <div>
    <div class="q-list q-list--bordered q-list--separator"></div>
    <div class="q-item q-item-type row no-wrap flex-center">
      <h5 class="text-grey-8">No Todos Item, try add some down here</h5>
    </div>
    <div class="q-item q-item-type row no-wrap column">
      <!---->
      <div class="full-width row">
        <div class="q-field row no-wrap items-start col-xs-8 col-md-10 q-input q-field--filled">
          <div class="q-field__inner relative-position col self-stretch column justify-center">
            <div tabindex="-1" class="q-field__control relative-position row no-wrap">
              <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip"><input tabindex="0" type="text" class="q-field__native"></div>
            </div>
          </div>
        </div> <button tabindex="0" type="button" class="q-btn inline q-btn-item non-selectable col-xs-4 col-md-2 flex-center q-btn--standard q-btn--rectangle q-focusable q-hoverable">
          <div tabindex="-1" class="q-focus-helper"></div>
          <div class="q-btn__content text-center col items-center q-anchor--skip justify-center row">Submit</div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
